pkg/controller/certificates/util.go

```diff
@@ -60,7 +60,7 @@ func PrivateKeyMatchesSpec(pk crypto.PrivateKey, spec cmapi.CertificateSpec) ([]
 func rsaPrivateKeyMatchesSpec(pk crypto.PrivateKey, spec cmapi.CertificateSpec) ([]string, error) {
 	rsaPk, ok := pk.(*rsa.PrivateKey)
 	if !ok {
-		return []string{"spec.keyAlgorithm"}, nil
```

Comment:
these are probably leftover from alpha/beta structure when the `privateKey` field did not exist yet